kubeadm/hack/publish-flannel.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/Masterminds/semver"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+const (
+	supportedVersions = ">= 0.12.0"
+)
+
+func dockerBuild(v *semver.Version, dockerfilePath string) error {
+	cmd := exec.Command("docker", "build", "--pull",
+		fmt.Sprintf("--build-arg=flannelVersion=%s", v.String()),
+		fmt.Sprintf("--tag=sigwindowstools/flannel:%s", v.String()),
```

Comment:
We should add our own version number appended to flannel.  Otherwise when we change files not related to the flannel version we will overwrite the tag with the newer version.

Reply:
Maybe you are right. I don't know what is better: build huge numbers of tags (one for future merge) or replace tags by updated version. For example, base windows tags replace previous tags when rollout new ones.

So, maybe it is not so bad to replace previous tags.

Reply:
The base windows tags don't replace the previous ones. They act as "latest" which point to specific image layer. Having a tag for each would make it so we are tied to a release and specific set of files.  We can also tag a latest to make it easier for on-boarding.

Reply:
Ok. I am going to add special suffix to image sigwindowstools/flannel.
For example: sigwindowstools/flannel:%flannelversion%-build%number%